### PR TITLE
bbolt be gone

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

File: pkg/osquery/extension.go [hunk 1]
```diff
@@ -600,20 +599,38 @@ func uint64FromByteKey(k []byte) uint64 {
...
-		return storage.StatusLogsStore.String(), nil
+		return s.StatusLogsStore(), nil
 	default:
-		return "", fmt.Errorf("unknown log type: %v", typ)
+		return nil, fmt.Errorf("unknown log type: %v", typ)
 
 	}
 }
 
+func sortStoreKeys(s types.KVStore) ([][]byte, error) {
```
[the remark this answers is not in the document]
Comment (directionless) [reply pkg/osquery/extension.go:615]: Hrm. [`ForEach`](https://pkg.go.dev/go.etcd.io/bbolt#Bucket.ForEach) is already an iterator. So I find I have questions...
1. Why do you need sort here?
2. Can we call ForEach directly
3. Our [`ForEach`](https://github.com/kolide/launcher/blob/main/pkg/agent/storage/bbolt/keyvalue_store_bbolt.go#L122) has a bug -- by the docs, it should be a RO transaction, not an RW one

File: pkg/agent/storage/inmemory/keyvalue_store_in_memory.go [hunk 2]
```diff
@@ -43,22 +42,44 @@ func (s *inMemoryKeyValueStore) Set(key, value []byte) error {
...
+		s.order = append(s.order, string(key))
+	}
+
+	s.setItem(key, value)
+	return nil
+}
+
+// setItem sets the item in the backing items map by copying the value to a
+// new slice
+func (s *inMemoryKeyValueStore) setItem(key, value []byte) error {
```
Comment (James-Pickett) [inline pkg/agent/storage/inmemory/keyvalue_store_in_memory.go:58]: drop error as return

File: pkg/osquery/extension.go [hunk 3]
```diff
@@ -662,51 +659,23 @@ func (e *Extension) writeLogsLoopRunner() {
...
-		for totalBytes := 0; k != nil; {
-			// A somewhat cumbersome if block...
-			//
-			// 1. If the log is too big, skip it and mark for deletion.
-			// 2. If the buffer would be too big with the log, break for
-			// 3. Else append it
-			//
-			// Note that (1) must come first, otherwise (2) will always trigger.
+	// Collect keys to delete
+	var keys [][]byte
```
Comment (James-Pickett) [inline pkg/osquery/extension.go:674]:
```suggestion
	var keysToDelete [][]byte
```

File: pkg/osquery/extension_test.go [hunk 4]
```diff
@@ -1350,3 +1365,21 @@ func Test_setVerbose_MalformedConfig(t *testing.T) {
 
 	require.Equal(t, malformedCfg, modifiedCfg)
 }
+
+// numberOfBufferedLogs returns the number of logs buffered for a given type.
+func (e *Extension) numberOfBufferedLogs(typ logger.LogType) (int, error) {
```
Comment (James-Pickett) [inline pkg/osquery/extension_test.go:1370]: should be able to just use store.Len() now

**Comment by James-Pickett** [comment]
closed in favor of https://github.com/kolide/launcher/pull/1652

**Pull request closed without merging**